R-package/DESCRIPTION

```diff
@@ -42,7 +42,11 @@ URL: https://github.com/Microsoft/LightGBM
 BugReports: https://github.com/Microsoft/LightGBM/issues
 NeedsCompilation: yes
 Biarch: true
+VignetteBuilder: 
+    knitr
 Suggests:
+    knitr,
+    rmarkdown,
     processx,
     testthat
```

Comment:
> > Would you stick to the same code style as in the package code? I am asking because LightGBM is the only project that use R code with comma first.
> 
> Yes please. And I really doubt we're the ONLY project using comma-first style haha

Actually I wanted to say "the only project I know" ;)

> > The vignette html will be created automatically when building the package. Since the whole building process of LGB is complex, I have no idea if this works out as I hope...
> 
> Could you elaborate on what you found complex about the process? On any system, this should work with no need to install anything else:
> 
> ```shell
> sh build-cran-package.sh
> R CMD install lightgbm*.tar.gz
> ```
Cool, I will try that!